Reverse conditional_derivative_decorator logic

kirkkwang opened the pull request:
ref #398 

This commit also address its spec however one of them do not pass and
will get addressed in a separated commit.

spec/extensions/hyrax/conditional_derivative_decorator_spec.rb:15 gives a false positve

spec/extensions/hyrax/conditional_derivative_decorator_spec.rb:21 does not pass

### I created an attachment w intermediate file type and attached a file set to it. The file set should render a thumbnail. ✅ 

![image](https://user-images.githubusercontent.com/10081604/228678859-2542b1c2-5f97-4d1f-bf9d-e193a65cb877.png)

![image](https://user-images.githubusercontent.com/10081604/228679108-5f667193-4e2c-4beb-b363-db15c3e19ec3.png)


### I created an attachment that's not an intermediate file type and attached a file set to it. The file set should NOT render a thumbnail. ✅ 

![image](https://user-images.githubusercontent.com/10081604/228679527-8ade0757-6e70-46ce-809e-82e604771ebf.png)

![image](https://user-images.githubusercontent.com/10081604/228679731-b4b5112f-ec0c-42e4-8d4c-be51aaff747a.png)

Commit:
Reserve conditional_derivative_decorator logic

This commit also address its spec however one of them do not pass and
will get addressed in a separated commit.

laritakr commented:
I'm confused about how this file relates to the derivative custom_derivative_service.

kirkkwang commented:
flip this logic too
https://github.com/scientist-softserv/utk-hyku/blob/5a8b7a6d7c95ff1dfcdf81533a84bc6cda08aba8/app/services/custom_derivative_service.rb#L6

Commit:
Merge branch 'main' into reverse-conditional-deriv-logic

Commit:
flip logic

ShanaLMoore commented:
> flip this logic too
> 
> https://github.com/scientist-softserv/utk-hyku/blob/5a8b7a6d7c95ff1dfcdf81533a84bc6cda08aba8/app/services/custom_derivative_service.rb#L6

✅ DONE

Commit:
reconcile gem updates

upgrade bulkrax, revert redlock. redlock is gem dependency of hyrax and the updated version causes errors; file_sets aren't able to attach to their works.

ShanaLMoore commented on app/services/custom_derivative_service.rb:
```diff
@@ -3,7 +3,7 @@
 module CustomDerivativeService
   def valid?
-    false if file_set.rdf_type&.join&.downcase&.include?("intermediate_file")
+    true if file_set.rdf_type&.join&.downcase&.include?("intermediate_file")
```
This doesn't seem to be getting called (yet), however, the logic needs to be updated nonetheless.

Commit:
Appease rubocop
